dll/win32/shell32/brfolder.cpp

```diff
@@ -1034,6 +1113,14 @@ BrFolderDlgProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
     if (!info)
         return 0;
 
+    if (info->pContextMenu)
+    {
+        LRESULT result;
+        if (S_OK == SHForwardContextMenuMsg(info->pContextMenu, uMsg, wParam,
+                                              lParam, &result, TRUE))
+            return TRUE | SetWindowLongPtr(hWnd, DWLP_MSGRESULT, result);
```

Comment:
TRUE or...

What is the meaning of this?

Reply:
The actual result is `TRUE | 0` but I really mean `TRUE | IDontCare`. Now go ahead and tell me this is not proper style because I know you want to...

Reply:
`TRUE | whatever` is going to bitwise-OR `1` with whatever `SetWindowLongPtr` returns.

Now, the actual question I have, is why don't you just return the value returned by `SetWindowLongPtr` ?
(The code as it currently is and the absence of comment doesn't explain the underlying reason why the developer, i.e. you, didn't want to do that.)

Reply:
> why don't you just return the value returned by `SetWindowLongPtr`

Because that is not how DWLP_MSGRESULT works. Sorry for putting it on the same line to mark them as belonging together, I will simplify.